### PR TITLE
Remove games.eclass

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
Status: temporarily on hold while I work on https://github.com/gentoo/gentoo/pull/19865 . Code reviews and comments still welcome, just please excuse any delay in action!

The intent of this PR is to update all required ebuilds such that the games eclass can be removed. I will leave it in `DRAFT` state until all the commits are ready, unless anyone thinks it would make more sense to submit them each individually.

Here is the list of packages that I am targetting:

- [ ] games-roguelike/hengband-1.6.2-r1
    - Ok, I've done some work on this but I'm struggling with the codebase and
      getting autotools to cooperate on the installation. I blame upstrem, not
      autotools. I plan to come back to this, but for now I will move on to the
      next one.
- [x] games-roguelike/adom-1.1.1-r1
    - also, bumped to version 3.3.3
- [ ] games-roguelike/FTL-1.5.13
- [ ] games-roguelike/zangband-2.7.4c
    - like hengband, another angband derivative. I got further with this one,
      but for some reason I can't figure out why DESTDIR isn't being respected.
      `games.eclass` made it work! why can't I?! We'll come back to this.
- [x] games-roguelike/tome-2.3.5
    - per https://bugs.gentoo.org/387061, time may be better spent packaging te4
      v1.7.2, which appears to have been released november 2020
- [x] dev-games/ps2-packer-0.4.4
- [x] dev-games/quake4-sdk-1.4.2
- [ ] games-arcade/bomns-0.99.2
- [ ] games-arcade/rockdodger-0.6.0a-r1
- [ ] games-arcade/dynamitejack-1.0.23-r1
- [ ] games-arcade/briquolo-0.5.7
- [ ] games-arcade/aquaria-1.1.3-r1
- [ ] games-arcade/betna-0.9.7
- [ ] games-arcade/burgerspace-1.9.2
- [ ] games-arcade/jardinains-2.0-r1
- [ ] games-arcade/blockrage-0.2.3
- [ ] games-arcade/emilia-pinball-0.3.1-r1
- [ ] games-arcade/pachi-1.0
- [ ] games-arcade/barbarian-bin-1.01-r1
- [ ] games-arcade/bumprace-1.5.3
- [ ] games-arcade/gunocide2ex-1.0
- [ ] games-arcade/cavezofphear-0.5.1
- [ ] games-arcade/epiar-0.5-r1
- [ ] games-mud/circlemud-3.1
- [ ] games-server/pvpgn-1.8.5-r2
- [ ] games-rpg/wasteland2-1.9.0.13
- [ ] games-rpg/dear-esther-20130608
- [ ] games-rpg/to-the-moon-0
- [ ] games-rpg/coe2-2007-r1
- [ ] games-strategy/ja2-stracciatella-0.12.1_p7072
- [ ] games-strategy/darwinia-1.4.0_beta9-r1
- [ ] games-strategy/dominions2-2.16
- [ ] games-strategy/ja2-stracciatella-data-1
- [ ] games-puzzle/splice-20121120
- [ ] games-puzzle/xwelltris-1.0.1
- [ ] games-puzzle/tiny-and-big-1.4.1
- [ ] games-puzzle/nightsky-20111222
- [ ] games-puzzle/seatris-0.0.14
- [ ] games-puzzle/xbomb-2.2b
- [ ] games-puzzle/xlogical-1.0.7
- [ ] games-puzzle/greedy-0.2.0-r1
- [ ] games-fps/quake2-data-3.20
- [ ] games-fps/sauerbraten-2013.02.03
- [ ] games-fps/quake1-killer-2.2z
- [ ] games-fps/quake1-demodata-1.06
- [ ] games-fps/ut2003-data-2107
- [ ] games-fps/etqw-bin-1.5-r2
- [ ] games-fps/qudos-0.40.1
- [ ] games-fps/quake2-demodata-3.14
- [ ] games-fps/quake1-data-2.40
- [ ] games-fps/ut2003-demo-2206-r4
- [ ] games-fps/quake1-textures-20050820
- [ ] games-fps/industri-1.01
- [ ] games-fps/quake2-textures-0_beta8
- [ ] games-fps/quake2-icculus-0.16.1-r1
- [ ] games-fps/red-blue-quake2-0.1
- [ ] games-fps/ut2003-bonuspack-epic-1
- [ ] games-fps/ezquake-bin-2.2
- [ ] games-fps/ut2003-bonuspack-cm-1
- [ ] games-fps/ut2003-bonuspack-de-1
- [ ] games-fps/tribes2-25034
- [ ] games-fps/ut2004-demo-3334-r1
- [ ] games-fps/quake1-rally-1.2
- [ ] games-fps/etqw-demo-2.0_p1-r2
- [ ] games-fps/quake1-teamfortress-2.9
- [ ] games-fps/soldieroffortune-1.06a-r1
- [ ] games-fps/glxquake-bin-0-r2
- [ ] games-action/hotline-miami-1.0.9a_p20140221-r1
- [ ] games-action/descent3-1.4.0b-r2
- [ ] games-action/shadowgrounds-bin-0_p1
- [ ] games-action/intrusion2-1.024
- [ ] games-action/solar2-1.10
- [ ] games-action/heretic2-1.06c-r1
- [ ] games-action/swordandsworcery-1.02
- [ ] games-action/heretic2-demo-1.06a
- [ ] games-action/towbowl-tactics-0.5
- [ ] games-action/shadowgrounds-survivor-bin-0_p1
- [ ] games-board/xscrabble-2.10-r2
- [ ] games-misc/papers-please-1.1.65
- [ ] games-misc/little-inferno-20130509
- [ ] games-misc/bsd-games-2.17-r5
- [ ] games-misc/games-envd-0

### CODE REVIEW

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:17]: This kind of works, although I'd prefer to not force this. Rebuilding ncurses with or without tinfo breaks ABI and it's kind of a pain point. If someone has it off, they may have a good reason (like some esoteric Prefix case or similar).

The solution is linked to my other comment coming...

File: games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch
Comment (thesamesam) [inline games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch:10]: I think we might technically need nursesw and so on (wide) for USE=unicode, which is now forced on. But in order to avoid forcing USE=tinfo (which is also forced on, but still), we should use PKG_CHECK_MODULES instead to determine the LIBS content based on what the system needs.

Comment (thesamesam) [reply games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch:10]: Here's an example: 4720158264396a3ffbe291018c5a8d81fa2fc31b.

**Review by thesamesam — CHANGES_REQUESTED** [review]
Left some specific comments re ```PKG_CHECK_MODULES```. If you need more of a hint, just give me a shout (as ever!)

Also, it's okay if you really want to just fix the ncurses bug, but let's try do EAPI 7 + no more games.eclass etc in a new commit too.

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild]: Removed it per our IRC convo.

File: games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch
Comment (ezzieyguywuf) [reply games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch]: Thanks for the example, I have fixed this to use `PKG_CHECK_MODULES`.

**Comment by ezzieyguywuf** [comment]
> Left some specific comments re `PKG_CHECK_MODULES`. If you need more of a hint, just give me a shout (as ever!)

Thanks as always @thesamesam !

I'm going to work on updating the ebuild to EAPI 7, then move on to the next on the list.

I'll plan to leave the `fixup` and `squashup` commits on the PR until tihs is ready to merge - I've worked on large PR's in the past where constantly rebasing made it difficult to navigate the PR itself.

Let me know if there's a preference to just go ahead and rebase and force push, though, b/c I can do that too.

**Comment by chewi** [comment]
Look out for existing PRs such as #17664.

**Comment by ezzieyguywuf** [comment]
> Look out for existing PRs such as #17664.

Nice one thanks! I'll make sure to search before starting a new package

File: games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch
Comment (SoapGentoo) [inline games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch:16]: `PKG_CHECK_MODULES([NCURSES], [ncurses], [LIBS="${LIBS} ${NCURSES_LIBS}"])`

Comment (ezzieyguywuf) [reply games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch:16]: Nice catch @SoapGentoo I'll fix this

File: games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch
Comment (thesamesam) [inline games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch:17]: Remove this when you're done, of course.

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:5]: Sort, EAPI 7, but you know that I hope ;)

Comment (thesamesam) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild:5]: (Especially given you're already using BDEPEND!!)

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:8]: Down below, sort to the canonical 'block structure':
LICENSE
SLOT
KEYWORDS
IUSE

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:59]: It seems odd to be able to drop this but not have to do some fiddling with paths in the build system

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:47]: Did egamesconf do any other magic for us?

Note that the [documentation](https://devmanual.gentoo.org/eclass-reference/games.eclass/index.html) says:
```
egamesconf [<args>...]
Games equivalent to 'econf' for autotools based build systems. It passes the necessary games specific directories automatically.
```

Looking at the [source](https://gitweb.gentoo.org/repo/gentoo.git/tree/eclass/games.eclass#n131), we can see it does a lot:
```
       [...]

	econf \
		--prefix="${GAMES_PREFIX}" \
		--libdir="$(games_get_libdir)" \
		--datadir="${GAMES_DATADIR}" \
		--sysconfdir="${GAMES_SYSCONFDIR}" \
		--localstatedir="${GAMES_STATEDIR}" \
		${_gamesconf} \
		"$@"
```


So, TL;DR: don't rush! :)

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch
Comment (ezzieyguywuf) [reply games-roguelike/hengband/files/hengband-1.6.2-autoconf-ncurses.patch:17]: Yup, this was just an oversight whenever i've been tweaking this patch, must have left it in on accident.

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild:5]: Definitely! I've been trying to attack the updates one at a time, with the eapi upgrade scheduled next (including adding `PATCHES` variable)

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild:59]: Yes i thought it was odd too! I think you're right that the answers probably lie herein

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild:47]: I actually had checked this, but based on what i understand about the whole "getting rid of games.eclass" thing, game packages are supposed to be just normal packages now.

I checked, and autotool.eclass seems to set some sane values for these, but to your point there's probably something games.eclass was doing for me before that's no longer being done.

So I'll take a closer look and see what I find :)

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild:59]: hrm, the default should do `make DESTDIR=${D} install`, which should enough to install this.

There's something fishy going on with `--with-setgid` config parameter, as it seems this needs to be used in order for installation to do anything.

I'm going to step away from this ebuild for a bit and come back with fresh eyes, perhaps working on some other  ones will help illuminate for me what I'm missing here.

File: games-roguelike/hengband/files/hengband-1.6.2-ovflfix.patch
Comment (thesamesam) [inline games-roguelike/hengband/files/hengband-1.6.2-ovflfix.patch:1]: I normally drop the lines before ---/+++

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:5]: New line after EAPI

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:23]:
```suggestion
PATCHES=(
    "${FILESDIR}/${PN}-1.6.2-added_faq.patch"
```

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:25]:
```suggestion
	"${FILESDIR}/${PN}-1.6.2-ovflfix.patch"
)
```

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:46]: gamestat [instead](https://projects.gentoo.org/qa/policy-guide/filesystem.html#pg0205)? (You'll need to then depend on it [appropriately](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-720008.1))

File: games-roguelike/tome/files/tome-2.3.5-format.patch
Comment (thesamesam) [inline games-roguelike/tome/files/tome-2.3.5-format.patch:1]: I normally [drop](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html#clean-patch-howto) the dates

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:4]: New line after EAPI. Please also fix the copyright headers (use repoman commit or pkgdev commit).

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:31]: adjust formatting

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:29]:
```suggestion
S="${WORKDIR}"/tome-${MY_PV}-src/src
```

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:43]: Not ready? :)

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:46]: Missing || dies. We always want || dies on external commands.

Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:46]: Below, there's a direct pkg-config call, but we want to use the helper function from toolchain.eclass.

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:37]: || die

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:114]: || die and gamestat maybe?

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [inline games-roguelike/tome/tome-2.3.5.ebuild:5]: Is anything using multilib.eclass here? get_libdir is commonly the reason but that's [part of](https://blogs.gentoo.org/mgorny/2015/11/13/the-ultimate-guide-to-eapi-6/) >= EAPI 6

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (thesamesam) [inline games-roguelike/zangband/zangband-2.7.4c.ebuild:52]: all in one line or new line after/before ()

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (thesamesam) [inline games-roguelike/zangband/zangband-2.7.4c.ebuild:59]: ${ED} in the latter case at least and.. || die

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: games-roguelike/adom/adom-1.1.1-r1.ebuild
Comment (mm1ke) [inline games-roguelike/adom/adom-1.1.1-r1.ebuild:28]: `edos2unix` requires the `edos2unix` eclass. (was before implicit inherit by `eutils`)

File: dev-games/ps2-packer/ps2-packer-0.4.4.ebuild
Comment (mm1ke) [inline dev-games/ps2-packer/ps2-packer-0.4.4.ebuild:1]: didn't use `repoman commit` ? ;)

File: games-roguelike/adom/adom-1.1.1-r1.ebuild
Comment (mm1ke) [inline games-roguelike/adom/adom-1.1.1-r1.ebuild:1]: didn't use `repoman commit` ? ;)

File: games-roguelike/adom/adom-3.3.3.ebuild
Comment (mm1ke) [inline games-roguelike/adom/adom-3.3.3.ebuild:1]: didn't use `repoman commit` ? ;)

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (mm1ke) [inline games-roguelike/hengband/hengband-1.6.2-r1.ebuild:5]: is `eutils` even required here?

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (mm1ke) [inline games-roguelike/zangband/zangband-2.7.4c.ebuild:6]: `multilib` shouldn't be required here too.

File: games-roguelike/adom/adom-3.3.3.ebuild
Comment (thesamesam) [reply games-roguelike/adom/adom-3.3.3.ebuild:1]: pkgcheck scan --commits / pkgdev commit would handle this too. Don't commit manually, use either pkgdev or repoman.

File: games-roguelike/hengband/hengband-1.6.2-r1.ebuild
Comment (thesamesam) [reply games-roguelike/hengband/hengband-1.6.2-r1.ebuild:5]: Always audit the list of eclasses for unused stuff and sort them.

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (thesamesam) [reply games-roguelike/zangband/zangband-2.7.4c.ebuild:6]: Ditto: always audit the list of eclasses for unused inherits and sort (unless it affects phases).

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (thesamesam) [reply games-roguelike/zangband/zangband-2.7.4c.ebuild:59]: To be clear: you need to always || die on external commands. rm, rmdir, anything that's not provided by PMS, and almost all eclass functions.

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/zangband/zangband-2.7.4c.ebuild:52]: you mean the preference is:

```sh
src_install()
{
    # content
}
```

? I personally prefer the opening bracket on the same line from a stylistic standpoint, is there a strong reason why it has to start on its own newline?

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/tome/tome-2.3.5.ebuild:5]: Didn't realize it was builtin after EAPI 6, that's the reason I added it.

File: games-roguelike/adom/adom-3.3.3.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/adom/adom-3.3.3.ebuild:1]: ok let's make a deal - how about I go ahead and set up some git hooks that runs `pkgcheck --scan --commits` prior to pushing anything.

I *want* to want to use `pkgdev commit`, but it truly just does not feel good to me.

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (thesamesam) [reply games-roguelike/zangband/zangband-2.7.4c.ebuild:52]: Sorry, I was referring to preferred array syntax:
DOCS=( )
or
DOCS=(

)

File: games-roguelike/tome/tome-2.3.5.ebuild
Comment (thesamesam) [reply games-roguelike/tome/tome-2.3.5.ebuild:5]: Now you know! (This is quite a common thing to have to fix, actually)

File: games-roguelike/adom/adom-3.3.3.ebuild
Comment (thesamesam) [reply games-roguelike/adom/adom-3.3.3.ebuild:1]: That works for me, but I recommend you run pkgcheck scan --commits before you're ready to push ;)

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/zangband/zangband-2.7.4c.ebuild:52]: Ah hah, closing paren on its own line.

Yea I've actually seen it both ways so wasn't quite sure i there was a preference.

Perhaps a "style guide" *would* be helpful, as you've suggested before...

File: games-roguelike/adom/adom-3.3.3.ebuild
Comment (ezzieyguywuf) [reply games-roguelike/adom/adom-3.3.3.ebuild:1]: >  I recommend you run pkgcheck scan --commits before you're ready to push ;)

Maybe I'll add it as a pre-commit hook then :)

File: games-roguelike/zangband/zangband-2.7.4c.ebuild
Comment (ionenwks) [reply games-roguelike/zangband/zangband-2.7.4c.ebuild:52]: > Perhaps a "style guide" _would_ be helpful, as you've suggested before...

There's a basic proxy-maint [Style Guide](https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide) but there's plenty it doesn't cover. Not like majority of style issues are enforced (outside the few things in the policy guide), and maintainers may have their own preferences. Mostly comes down to the most common usage so it's less all over the place.

**Comment by ezzieyguywuf** [comment]
@thesamesam should this be closed now?

**Comment by thesamesam** [comment]
I think so, but we can cherry-pick anything if I missed stuff

**Pull request closed without merging**
